src/Uno.UI/UI/Xaml/ResourceDictionary.cs

```diff
@@ -226,10 +260,62 @@ public void CopyTo(global::System.Collections.Generic.KeyValuePair<object, objec
 
 		public bool IsReadOnly => false;
 
+		private bool _isParsing;
+		/// <summary>
+		/// True if the element is in the process of being parsed from Xaml.
+		/// </summary>
+		/// <remarks>This property shouldn't be set from user code. It's public to allow being set from generated code.</remarks>
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public bool IsParsing
+		{
+			get => _isParsing;
+			set
+			{
+				if (!value)
+				{
+					throw new InvalidOperationException($"{nameof(IsParsing)} should never be set from user code.");
+				}
+
+				_isParsing = value;
+				if (_isParsing)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this condition so that it does not always evaluate to 'true'.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826539)